src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/RotatingStar.cpp

```diff
@@ -636,7 +714,9 @@ tuples::TaggedTuple<hydro::Tags::Pressure<DataType>> RotatingStar::variables(
     tmpl::list<hydro::Tags::Pressure<DataType>> /*meta*/) const {
   const auto rest_mass_density = get<hydro::Tags::RestMassDensity<DataType>>(
       variables(vars, x, tmpl::list<hydro::Tags::RestMassDensity<DataType>>{}));
-  return {equation_of_state_.pressure_from_density(rest_mass_density)};
+  using std::max;
+  return {equation_of_state_->pressure_from_density(
+      Scalar<DataType>{DataType{max(1.0e-300, get(rest_mass_density))}})};
```

Comment:
Some EoS's run into FPEs when fed in 0 for these kinds of state variable functions (in my case it was `Spectral`). Likely it wasn't needed before because people were only running with polytropes.
I'm happy to make it neater as you suggest, but I should note that I just took this approach from elsewhere in the file (see line 693). I'm not sure why it was there before, but I'm assuming for similar reasons? It also shows up in line 507 of `src/PointwiseFunctions/AnalyticData/GrMhd/CcsnCollapse.cpp`. So this may speak to a broader issue, e.g. maybe we should put protection from FPEs in the case of 0 density directly in these EoS functions instead of what I'm doing here to compensate for it externally

Reply:
I don't like the idea of protecting against 0 in the EOS because practically speaking we need an atmosphere anyway since the evolution equations assume a large number of particles, and, more importantly, because this adds checks that are done every step of the evolution where we have other code that guarantees positive definite numbers. I think unfortunately the best approach is what you're doing here. We override the small numbers later anyway with our atmosphere fix :shrug: I do wonder if 1e-50 would be better since in some cases (e.g. polytrope) we might square the number and 1e-300^2=underflow and that can cause problems